src/Microsoft.DotNet.GenAPI/Shared/HeaderFile.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+
+namespace Microsoft.DotNet.GenAPI.Shared;
+
+public class FileHeader
+{
+    public static readonly string DefaultFileHeader = """
+        //------------------------------------------------------------------------------
+        // <auto-generated>
+        //     This code was generated by a tool.
+        //     Roslyn-based GenAPI Version: {0}
+        //
+        //     Changes to this file may cause incorrect behavior and will be lost if
+        //     the code is regenerated.
+        // </auto-generated>
+        //------------------------------------------------------------------------------
+        """;
+
+    public static string ReadFromFile(string? headerFile)
+    {
+        if (headerFile != null && !string.IsNullOrEmpty(headerFile))
+        {
+            return File.ReadAllText(headerFile);
+        }
+
+        return DefaultFileHeader;
```

Comment:
`IsNullOrEmpty` is enough:

```suggestion
        if (!string.IsNullOrEmpty(headerFile))
        {
            return File.ReadAllText(headerFile);
        }

        return DefaultFileHeader;
```

or possibly even:
```
        return !string.IsNullOrEmpty(headerFile)
            ? File.ReadAllText(headerFile)
            : DefaultFileHeader;
```